lemmatizer/lemmatizer_2.py

```diff
@@ -0,0 +1,127 @@
+from __future__ import absolute_import
+from __future__ import print_function
+
+import os
+import sys
+import dill
+import cPickle
+
+sys.path.append('../common')
+from util import batch_index_generator, split_data
+import activation
+import cosine_mlp, thin_cosine_mlp
+
+# to generate a training dataset
+import numpy as np
+from gensim import models
+from nltk import pos_tag, word_tokenize
+from nltk.corpus import wordnet
+from nltk.stem import WordNetLemmatizer
+
+
+def read_data_from_file(file):
+    with open(file) as fp:
+        data = fp.read()
+    data = data.split('\r\n')
+    data = [i for i in data if len(i) > 0]
+
+    num_data = len(data)
+    model = models.Word2Vec.load_word2vec_format(
+        '../storage/pos_tagger/GoogleNews-vectors-negative300.bin',
+        binary=True)
+    vectorizer = lambda x: model[x] if x in model else np.zeros(300)
+
+    X = np.zeros((num_data, 300))
+    y = np.zeros((num_data, 300))
+
+    all_words = []
+    for i in range(len(data)):
+        lemma, word = data[i].split('\t')
+        all_words.append(word)
+        word_vec = vectorizer(word)
+        lemma_vec = vectorizer(lemma)
+
+        X[i, :] = word_vec
+        y[i, :] = lemma_vec
+
+    word_pos = [p[1]for p in pos_tag(all_words)]
+    pos_set = list(set(word_pos))
+    pos_encoder = one_hot_encoding(pos_set)
```

Comment:
In ML, you want to predict a relationship between inputs and outputs. These outputs are often discrete, so for example, outputs could a single digit between 1 and 10, like 1 or 4. You can imagine the neural networks trying to pick a "class" between 1 and 10 for every input. This is often used in image recognition stuff or handwriting recognition. So... when training the neural network, we usually don't just feed the network 4 or 10, we instead "encode" them in an array. 

What a one-hot-encoding is for example (1-indexing of arrays):
1 --> [1, 0, 0, 0, 0, 0, 0, 0, 0, 0]
4 --> [0, 0, 0, 1, 0, 0, 0, 0, 0, 0]
8 --> [0, 0, 0, 0, 0, 0, 0, 1, 0, 0]

The point it to better represent the space for the neural net so it better understands the shape of the outputs.